include/fiction/utils/layout_utils.hpp

```diff
@@ -408,42 +415,71 @@ CoordinateType random_coordinate(CoordinateType coordinate1, CoordinateType coor
  * @param cell_se The southeast SiQAD cell defining the ending point of the area.
  * @return A vector containing all cells within the specified area.
  */
-[[nodiscard]] inline std::vector<siqad::coord_t> all_sidbs_in_spanned_area(const siqad::coord_t& cell_nw,
-                                                                           const siqad::coord_t& cell_se) noexcept
+template <typename CoordinateType>
+[[nodiscard]] inline std::vector<CoordinateType> all_sidbs_in_spanned_area(const CoordinateType& cell_nw,
```

Comment:
This function doesn't require SiDBs anywhere. Can we maybe change its name to reflect that?